### PR TITLE
refactor(transformer/decorators):  if it is a private method definition, transform it

### PR DESCRIPTION
**Pull request by Dunqing**

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2427?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #2426** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2427** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2427?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/2427?utm_source=stack-comment).

File: crates/oxc_transformer/src/proposals/decorators.rs [hunk 1]
```diff
@@ -391,10 +403,71 @@ impl<'a> Decorators<'a> {
...
+                                    SPAN,
+                                    self.ast.new_vec(),
+                                    self.ast.new_vec_single(
+                                        self.get_call_with_arguments(name.clone(), arguments),
+                                    ),
+                                )),
+                                self.ast.copy(&def.value.type_parameters),
+                                self.ast.copy(&def.value.return_type),
+                                self.ast.copy(&def.value.modifiers),
+                            );
```
Comment (Dunqing) [inline crates/oxc_transformer/src/proposals/decorators.rs:467]: Here, we need to clone the function and modify the parameters and body. It seems that we need a clone function in `AstBuilder` if it is possible.

Comment (Boshen) [reply crates/oxc_transformer/src/proposals/decorators.rs:467]: Can we hack it somehow. Why is memcpy a nightly function lol https://doc.rust-lang.org/std/mem/fn.copy.html

Comment (Dunqing) [reply crates/oxc_transformer/src/proposals/decorators.rs:467]: Looks like memcpy cannot solve. The copy function needs a copy trait. I thought we could add a `copy_function` function. This function implementation is like `move_expression`. create an empty function and replace it. What do you think?

**Comment by Boshen** [comment]
Let's rebase and merge first.

**Pull request merged**